src/content.js

```diff
@@ -115,7 +117,19 @@ async function buildMetaIndex(contentDir, tagsDir, baseUrl) {
     ])
     .join("\n");
 
-  return {pages, mdFooter, tags, baseUrl};
+  const findTagPageByName = (tagName) => {
+    const page = pages.find(page => page._slug == tagName && page.template == "tag");
+    if (!page) throw new Error(`Failed to find tag page for ${tagName}. Please create one`);
+    return page;
+  };
+
+  return {
+    findTagPageByName,
+    pages,
+    mdFooter,
+    data,
+    baseUrl
+  };
```

Comment:
This object can be thought of as a site-level API available to any page template as `metaIndex`. It offers cross-page indexing information, structured data like tag definitions, and some helper functions like the newly added `findTagPageByName`.